File: 5-longest-palindromic-substring/longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        int resLen = 0;
        String res = "";
        int n= s.length();
        for(int i = 0;i<n;i++){
            int low = i;
            int high = i;

            while(low >= 0 && high<n && (s.charAt(low) == s.charAt(high))){
                if(resLen < high - low +1 ){
                    res = s.substring(low,high+1);
                    resLen = Math.max(resLen,high - low +1);
                }
                low--;
                high++;
            }
            low = i;
            high = i+1;

            while(low >= 0 && high<n && (s.charAt(low) == s.charAt(high))){
                if(resLen < high - low +1 ){
                    res = s.substring(low,high+1);
                    resLen = Math.max(resLen,high - low +1);
                }
                low--;
                high++;
            }
        }
        return res;
    }
}

File: 5-longest-palindromic-substring/longest-palindromic-substring.c
char* longestPalindrome(char* s) {
    int n=strlen(s);
    if(n==0){
        return "";
    }
        int st=0;
        int end=0;
        for(int i=0;i<n;i++){
            int low=i;
            int high=i;
            while(low >=0 && high <n && s[low]==s[high]){
                if(end-st+1 < high-low+1){
                    end=high;
                    st=low;
                }
                low--;
                high++;
            }

            low=i;
            high=i+1;
            while(((low >=0) && (high <n)) && (s[low]==s[high])){
                if(end-st+1 < high-low+1){
                    end=high;
                    st=low;
                }
                low--;
                high++;
            }
        }
        int len = end - st + 1;
        char* ans=(char*)malloc(len+1);
        strncpy(ans,s+st,len);
        ans[len] = '\0';
        return ans;
}